server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -95,11 +95,12 @@ default Repository create(RepositoryMetaData metaData, Function<String, Reposito
     /**
      * Returns the index metadata associated with the snapshot.
      *
+     * @param repositoryData current {@link RepositoryData}
      * @param snapshotId the snapshot id to load the index metadata from
      * @param index      the {@link IndexId} to load the metadata from
      * @return the index metadata about the given index for the given snapshot
      */
-    IndexMetaData getSnapshotIndexMetaData(SnapshotId snapshotId, IndexId index) throws IOException;
+    IndexMetaData getSnapshotIndexMetaData(RepositoryData repositoryData, SnapshotId snapshotId, IndexId index) throws IOException;
```

Comment:
Kind of an awkward API to pass the `RepositoryData` here again admittedly, but as you can see from the diff we have the `RepositoryData` available 100% of the time in production code when calling this method so I didn't see a reason to make this call more expensive, slower and less stable by forcing another round trip to loading `repositoryData` just so that I can keep the simpler API.